### PR TITLE
Feature/e commerce protos Roberto Pérez

### PR DESCRIPTION
**Pull request by robert-z**

### CODE REVIEW

File: staff/roberto-perez/e-commerce-protos/scripts/cart.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+Cart.prototype.add = function(product) {
+  if (!(product instanceof Product))
+    throw new TypeError(product + " is not a product");
+
+  return this.products.push(product);
+};
+
+Cart.prototype.totalPrice = function() {
+  return this.products.reduce(function(accum, product) {
+    if (!product.price) product.price = 0;
```
Comment (maryammdot) [inline staff/roberto-perez/e-commerce-protos/scripts/cart.js]: In fact an instance of a product should always have price, so in my opinion this validations is unnecessary.

File: staff/roberto-perez/e-commerce-protos/scripts/cart.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    return accum + product.price;
+  }, 0);
+};
+
+Cart.prototype.numberOfItems = function() {
+  return this.products.length;
+};
+
+Cart.prototype.mostExpensive = function() {
```
Comment (maryammdot) [inline staff/roberto-perez/e-commerce-protos/scripts/cart.js:22]: cool aplication of reduce method!! 🤓

File: staff/roberto-perez/e-commerce-protos/scripts/cart.js (hunk 1 above)
Comment (robert-z) [reply staff/roberto-perez/e-commerce-protos/scripts/cart.js]: Done!

File: staff/roberto-perez/e-commerce-protos/scripts/cart.js (hunk 2 above)
Comment (robert-z) [reply staff/roberto-perez/e-commerce-protos/scripts/cart.js:22]: Thx! :relaxed:

**Pull request merged**